### PR TITLE
Performance improvements, modulo and Exception fix, minor refactoring

### PR DESCRIPTION
**Pull request by AnErrupTion**
This PR makes the Clt, Clt_Un, Cgt, Cgt_Un and Ceq opcodes use SetByteOnCondition(), which replaces the manual branching that had to be done originally.
It also introduces minor refactoring, like removing unused parentheses and converting fields to auto-properties.

### CODE REVIEW

**Comment by AnErrupTion** [comment]
I've pushed some more changes including a fix for modulo on doubles specifically, and a temporary fix which adds an infinite loop if an exception wasn't thrown.

File: source/Cosmos.IL2CPU/IL/Cgt.cs [hunk 1]
```diff
@@ -101,20 +101,20 @@ public override void Execute(Il2cpuMethodInfo aMethod, ILOpCode aOpCode)
...
-					XS.Jump(ConditionalTestEnum.LessThan, LabelTrue);
-					XS.Jump(LabelFalse);
-					XS.Label(LabelTrue);
-					XS.Add(XSRegisters.ESP, 4);
-					XS.Push(1);
-					XS.Jump(xNextLabel);
-					XS.Label(LabelFalse);
-					XS.Add(XSRegisters.ESP, 4);
-					XS.Push(0);
+                    /*XS.Xor(EBX, EBX);
```
Comment (quajak) [inline source/Cosmos.IL2CPU/IL/Cgt.cs:106]: Please remove commented out code

**Review by quajak — CHANGES_REQUESTED** [review]
Thank you for the improvements! Just one small change and othewise it looks good. Regarding the exception handling, more work is needed but that is for another pr to fix. 

Could you please make a small PR to the Cosmos repo to add a test to ensure that modulo now works correctly?

**Comment by AnErrupTion** [comment]
> Could you please make a small PR to the Cosmos repo to add a test to ensure that modulo now works correctly?

Will do soon!

**Review by quajak — APPROVED** [review]

**Pull request merged**